tun/tun_linux.go

```diff
@@ -399,6 +491,77 @@ func (tun *NativeTun) Close() error {
 	return err2
 }
 
+func (tun *NativeTun) BatchSize() int {
+	return tun.batchSize
+}
+
+func (tun *NativeTun) DisableOffload() error {
+	sc, err := tun.File().SyscallConn()
+	if err != nil {
+		return err
+	}
+	var ierr error
+	if err := sc.Control(func(fd uintptr) {
+		ierr = unix.IoctlSetInt(int(fd), unix.TUNSETOFFLOAD, ^tunOffloads)
+	}); err != nil {
+		return err
+	}
+
+	if ierr == nil {
+		tun.batchSize = 1
+	}
+	return ierr
+}
+
+// TUN offload features (TUNSETOFFLOAD ioctl) are defined in the kernel in
+// include/uapi/if_tun.h. The kernel symbols begin with TUN_F_.
+const (
+	tunOffloadFeatureChecksum = 0x01
+	tunOffloadFeatureTSO4     = 0x02
+	tunOffloadFeatureTSO6     = 0x04
+	tunOffloadFeatureTSOECN   = 0x08
```

Comment:
Again, any potential for x/sys/unix with all these constants? (I realize the tun package typically hasn't relied on that, so mostly just wondering.)

Reply:
yes, assuming that x/sys/unix is happy to take on these apis

Reply:
https://go-review.googlesource.com/c/sys/+/468656

Reply:
Updated in https://github.com/WireGuard/wireguard-go/pull/64/commits/ba7240c56fef5599358a394511813370be195cfa